src/gsy_e/models/myco_matcher/myco_internal_matcher.py

```diff
@@ -67,6 +67,7 @@ def match_recommendations(self, **kwargs):
                     continue
                 while True:
```

Comment:
I feel like I asked this question already before: Why is this `while True` loop here in the first place? Wouldn't calling the `self._get_matches_recommendations()` and `market.match_recommendations()` without it also work?
I think it is worth to add an inline comment about it.

Reply:
I am not sure, but to my eyes the rationale behind this is to iterate this process multiple times, until the available recommendations from residual orders are exhausted. For instance, if the first recommendation set produces trades, then some residual orders will be created which might be potential matches, therefore the process has to be repeated in order to match those etc. I'll reflect this on a comment. 
Without this fix, the problem was that if no trade was performed in the market.match_recommendations call, then the _get_matches_recommendations would return exactly the same pairs, which would then not be matched by market.match_recommendations, thus running to infinity.

Reply:
I understand now, Thanks a lot! Can you maybe add a small comment here, e.g. `Perform matching until all recommendations and their residuals are handled.` (there is a probably a better way to formulate this) . Thanks!

Reply:
I think I will go with your recommendation, sounds clear enough to me. Thanks!